chsdi/views/catalog.py

```diff
@@ -24,6 +24,7 @@ def tree_data(G, root, attrs, meta):
         raise nx.NetworkXError('Attribute names are not unique.')
 
     def add_children(n, G):
+        distinct_order_keys = set()
         order_key = lambda x: x['orderKey']
         label_key = lambda x: x['label'].replace(u'Ü', u'U').replace(u'Ä', u'A')
```

Comment:
I think I did try something like that, I recall it didn't work out, but feel free to address this weak point.